### PR TITLE
Fix vulnerability

### PR DESCRIPTION
**Pull request by LiilyZhang**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,64 +1,92 @@
 module github.com/open-horizon/vault-exchange-auth
 
-go 1.19
+go 1.21
```
Comment (bencourliss) [inline go.mod:3]: If you change this, you'll need to change the GitHub Action as well, https://github.com/open-horizon/vault-exchange-auth/blob/9590a05453f68e1df1960ec1eb47802c1943bdfe/.github/workflows/build-push.yml#L93-L97

File: go.mod [hunk 2]
```diff
@@ -1,64 +1,92 @@
...
 
 require (
-	github.com/hashicorp/vault/api v1.9.1
-	github.com/hashicorp/vault/sdk v0.9.0
+	github.com/hashicorp/vault/api v1.13.0
```
Comment (bencourliss) [inline go.mod:6]: @naphelps are we ok to increment the API and SDK to these versions? I'm not sure how these correlate to the actual Vault v1.14.x versions we're allowed to use.

**Review by bencourliss — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (naphelps) [reply go.mod:6]: @bencourliss I checked the licenses for all Hashicorp Golang modules that were updated. All of the licenses were allowed MPL 2.0 licenses.

**Review by naphelps — DISMISSED** [review]

**Comment by bencourliss** [comment]
@LiilyZhang Can you also change the [Vault version](https://github.com/open-horizon/vault-exchange-auth/blob/9590a05453f68e1df1960ec1eb47802c1943bdfe/Makefile#L7) to `1.14.10` which I believe is the latest and also the [version](https://github.com/open-horizon/vault-exchange-auth/blob/9590a05453f68e1df1960ec1eb47802c1943bdfe/Makefile#L13) of this container to `1.1.4`?

**Comment by naphelps** [comment]
> @LiilyZhang Can you also change the [Vault version](https://github.com/open-horizon/vault-exchange-auth/blob/9590a05453f68e1df1960ec1eb47802c1943bdfe/Makefile#L7) to `1.14.10` which I believe is the latest and also the [version](https://github.com/open-horizon/vault-exchange-auth/blob/9590a05453f68e1df1960ec1eb47802c1943bdfe/Makefile#L13) of this container to `1.1.4`?

This needs to be `<= v1.14.8`!

**Review by bencourliss — APPROVED** [review]

**Pull request merged**